sys/drv/stdvga.c

```diff
@@ -122,47 +122,38 @@ static int stdvga_set_fbinfo(stdvga_state_t *vga, fb_info_t *fb_info) {
   return 0;
 }
 
-static int stdvga_open(vnode_t *v, int mode, file_t *fp) {
-  stdvga_state_t *vga = devfs_node_data(v);
+static int stdvga_open(devnode_t *dev, file_t *fp, int oflags) {
+  stdvga_state_t *vga = dev->data;
   int error;
 
   /* Disallow opening the file more than once. */
   int expected = 0;
   if (!atomic_compare_exchange_strong(&vga->usecnt, &expected, 1))
     return EBUSY;
 
-  /* On error, decrease the use count. */
-  if ((error = vnode_open_generic(v, mode, fp)))
-    goto fail;
-
-  if ((error = stdvga_set_fbinfo(vga, &vga->fb_info)))
-    goto fail;
+  if ((error = stdvga_set_fbinfo(vga, &vga->fb_info))) {
+    atomic_store(&vga->usecnt, 0);
+    return error;
+  }
 
   return 0;
```

Comment:
if you return `error` above, maybe you should do the same here?

Reply:
It can be easily proven that the function may only return 0 here. We prefer to exit early with an error and reach final return only when the function performed its action successfully.